index.js

```diff
@@ -272,8 +272,8 @@ function session(options) {
           return ret;
         }
 
-        if (!res._header) {
-          res._implicitHeader()
+        if (typeof res.headersSent === 'boolean' ? !res.headersSent : !res._header) {
```

Comment:
Maybe it's better to make it a function like in compression https://github.com/expressjs/compression/pull/129